### ISSUE TITLE
Mixed precision tutorial does not work for custom training loop

### ISSUE THREAD

**Issue by CaptainDario** [keras-team/keras#20894]
I am trying to follow the [mixed precision tutorial](https://www.tensorflow.org/guide/mixed_precision) when using a custom training loop, specifically

```python
@tf.function
def train_step(x, y):
  with tf.GradientTape() as tape:
    predictions = model(x)
    loss = loss_object(y, predictions)
    scaled_loss = optimizer.get_scaled_loss(loss)
  scaled_gradients = tape.gradient(scaled_loss, model.trainable_variables)
  gradients = optimizer.get_unscaled_gradients(scaled_gradients)
  optimizer.apply_gradients(zip(gradients, model.trainable_variables))
  return loss
```

When trying to invoke, `get_scaled_loss` and `get_unscaled_gradients` an exception is thrown that those methods are undefined.

Maybe this is related to https://github.com/keras-team/keras/issues/19244

**Comment by mehtamansi29** [comment]
Hi @CaptainDario -

Thanks for reporting the issue. Here for mixed precision tutorial when using custom training loop try to use keras with tensorflow backend. 
Here is the [gist](https://colab.sandbox.google.com/gist/mehtamansi29/e38ce09b4985f279fdb1fe6c814c80c0/20894-mixed-precision-tutorial-does-not-work-for-custom-training-loop.ipynb) shows custom training loop is running fine. 
For more details you can find the reference [here](https://keras.io/guides/writing_a_custom_training_loop_in_tensorflow/#writing-a-training-loop-from-scratch-in-tensorflow).

**Comment by CaptainDario** [comment]
Thank you for your reply.
Maybe I am missing something, but I do not see how your answer is using mixed precision.

**Comment by mehtamansi29** [comment]
Hi @CaptainDario -

In keras documentation you can find [mixed precision](https://keras.io/api/mixed_precision/) here. 
Here in the gist you can find the model using mixed precision where model  output will be float16.

**Comment by CaptainDario** [comment]
I still do not understand how this answers my question. Your gist does not even set the mixed precision.
The documentation states that during mixed precision training, the loss needs to be scaled. However, the provided tutorial does not work.
Therefore, I am not asking how to start training in mixed precision but how to scale the loss correctly.

**Comment by mehtamansi29** [comment]
Hi @CaptainDario -

In keras3, you can use [Loss Scale Optimizer](https://keras.io/api/optimizers/loss_scale_optimizer/) to train your model without custom training loop. 
Loss Scale Optimizer wrap `keras.optimizers.Optimizer` instance and using dynamic_growth_steps argument  can update the scale upwardsand also set initial scale as well. 
Attached [gist](https://colab.sandbox.google.com/gist/mehtamansi29/6a46480368ba3683b69434b5d434a285/loss-scale-optimizer.ipynb) here for the reference.

**Comment by CaptainDario** [comment]
thank you for your reply.
Can I just use it as a normal optimizer when using it in a custom training loop?

**Comment by mehtamansi29** [comment]
Yes @CaptainDario - you can use it as normal optimizer when using it custom training loop. 
[Here](https://keras.io/guides/writing_a_custom_training_loop_in_tensorflow/#a-first-endtoend-example) is example for custom training loop with normal optimizer.

**Comment by CaptainDario** [comment]
thank you!
